docs/getting-started/how-prisma-schemas-are-converted-into-entities.md

```diff
@@ -0,0 +1,283 @@
+
+### Step 2: Prepare   
+
+During the prepare phase, Amplication runs the schema through 5 different operations:
+
+1. prepareModelNames
+2. prepareFieldNames  
+3. prepareFieldTypes
+4. prepareModelIdAttribute
+5. prepareIdField
+
+Each operation represents a transformation to be made on the schema, and their collective execution ensures the schema is in the correct format before being used for Amplication’s entities and fields creation.
+
+#### Model Names
+
+This part renames models in the Prisma schema to follow Amplication's specific format.
+
+Amplication's format is model names in **PascalCase**, **no underscores**, and **singular**.
+
+All original model names are preserved in the database with the `@@map` custom attribute where necessary.
+
+:::note
+If the model already has the **`@@map`** attribute, it won’t be added, even if the model name was formatted. 
+:::
+
+Here's an example of a transformation that would happen on one model:
+
+```diff title="schema.prisma"
+-model course_rating  
++model CourseRating {
+  id        String    @id @default(cuid())
+	createdAt DateTime  @default(now())
+  updatedAt DateTime  @updatedAt
+	rating    Int      
+
++@@map("course_rating") 
+}
+```
+
+#### Field Names
+
+Field names will also go through a similar conversion.
+
+Amplication's format for field names is **camelCase** and **no underscores**.
+
+
+It also ensures that original field names are preserved in the database by adding `@map` custom attributes where needed.
+
+:::note
+If the field already has the **`@map`** attribute, it won’t be added, even if the field name was formatted.
+:::
+
+Here's a quick example of a simple field name transformation:
+
+```diff title="schema.prisma"
+model Product {
+  id          String   @id @default(cuid())
+  createdAt   DateTime @default(now())
+  updatedAt   DateTime @updatedAt
+  name        String?
+-item_price  Float?
++itemPrice  Float?    @map("item_price")
+  description String?
+  orders      Order[]
+}
+```
+
+#### Field Types
+
+Now that Amplication has updated both the model names and field names, it has to update the types of the fields.
+
+For example, here's how two models could see transformations:
+
+```diff name="schema.prisma"  
+-model products
++model Product {
+  id          String   @id @default(cuid())
+  createdAt   DateTime @default(now())
+  updatedAt   DateTime @updatedAt
+  name        String?
+  itemPrice   Float?
+  description String? 
+-orders      orders[]
++orders      Order[]
+}
+
+-model orders
++model Order {
+  id         String    @id @default(cuid())
+  createdAt  DateTime  @default(now())
+  updatedAt  DateTime  @updatedAt
+  totalPrice Int?
+-product    products?  @relation(fields: [productId], references: [id])  
++product    Product?  @relation(fields: [productId], references: [id])
```

Comment:
I think this entire section is not needed since it is not providing any value to the reader. Obviously when changing the name of the model we also need to change the reference to it - but as for the reader this is obvious and we don't need to mention that